companion/lib/Service/HttpApi.ts

```diff
@@ -641,4 +649,164 @@ export class ServiceHttpApi {
+					},
+				}),
+				{}
+			)
+
+		res.json({
+			success: true,
+			message: 'Connection information retrieved successfully',
+			timestamp: getTimestampUnix(),
+			data: {
+				connections,
+				instanceStatus,
+			},
+		})
+		return
+	}
+
+	/**
+	 * Provides JSON for module or custom variables
+	 *
+	 */
+	#variablesGetJson = (req: Express.Request, res: Express.Response): void => {
+		const connectionLabel = req.params.label
+		this.logger.debug(`Got HTTP /api/variables/${connectionLabel}/json`)
+
+		// Determine variable type and fetch definitions
+		const isCustomVariable = connectionLabel === 'custom'
+		const variableDefinitions = isCustomVariable
+			? this.#serviceApi.getCustomVariableDefinitions()
+			: this.#serviceApi.getConnectionVariableDefinitions(connectionLabel)
+
+		const result: Record<string, any> = {}
+
+		for (const [variableName, obj] of Object.entries(variableDefinitions)) {
+			const value = isCustomVariable
+				? this.#serviceApi.getCustomVariableValue(variableName)
+				: this.#serviceApi.getConnectionVariableValue(connectionLabel, variableName)
+
+			if (value !== undefined) {
+				result[variableName] = {
+					value,
+					...obj,
+				}
+			}
+		}
+
+		if (Object.keys(result).length == 0) {
+			res.status(404).json({
+				success: false,
+				message: 'No variables found for this connection',
+				timestamp: getTimestampUnix(),
+				data: {
+					connection: connectionLabel,
+				},
+			})
+			return
+		}
+
+		res.json({
+			success: true,
+			message: 'Variables retrieved successfully',
+			timestamp: getTimestampUnix(),
+			data: {
+				connection: connectionLabel,
+				variableCount: Object.keys(result).length,
+				variables: result,
+			},
+		})
+		return
+	}
+
+	/**
+	 * Provides Prometheus metrics for module or custom variables
+	 *
+	 */
+	#variablesGetMetrics = (req: Express.Request, res: Express.Response): void => {
+		const connectionLabel = req.params.label
+		this.logger.debug(`Got HTTP /api/variables/${connectionLabel}/metrics`)
+
+		// Determine variable type and fetch definitions
+		const isCustomVariable = connectionLabel == 'custom'
+		const variableDefinitions = isCustomVariable
+			? this.#serviceApi.getCustomVariableDefinitions()
+			: this.#serviceApi.getConnectionVariableDefinitions(connectionLabel)
+
+		let result = ''
+
+		for (const [variableName, obj] of Object.entries(variableDefinitions)) {
+			const variableNamePrint = variableName.replace(/-/g, '_')
+			const metricName = `${connectionLabel}:${variableNamePrint}`
+
+			const value = isCustomVariable
+				? this.#serviceApi.getCustomVariableValue(variableName)
+				: this.#serviceApi.getConnectionVariableValue(connectionLabel, variableName)
+
+			const labelPrint = (isCustomVariable ? obj.description : obj.label)
+				.replace(/\\/g, '\\\\')
+				.replace(/"/g, '\\"')
+				.replace(/\n/g, '\\n')
+
+			result += `# HELP ${connectionLabel}:${variableNamePrint} ${labelPrint}\n`
```

Comment:
Personally, I would be tempted to use https://github.com/siimon/prom-client to build the response of this request, to avoid needing to manually handle this encoding and syntax structure.  
I don't know if it can do what we need here though, I have only used it for an application wide endpoint.